### PR TITLE
Provide Bazel environment variables for launchable schemes

### PR DESCRIPTION
**Pull request by cgrindel**
- Provide `BUILD_WORKSPACE_DIRECTORY` and `BUILD_WORKING_DIRECTORY` for launchable schemes.

### CODE REVIEW

File: tools/generator/test/CreateXCSchemesTests.swift [hunk 1]
```diff
@@ -179,6 +183,13 @@ fi
             file: file,
             line: line
         )
+        XCTAssertEqual(
+            testAction.customLLDBInitFile,
+            expectedCustomLLDBInitFile,
+            "testAction.customLLDBInitFile did not match for \(scheme.name)",
+            file: file,
+            line: line
+        )
```
Comment (cgrindel) [inline tools/generator/test/CreateXCSchemesTests.swift:192]: This was grouped with the launch action assertions. I just moved it to be with the test action assertions.

File: tools/generator/src/XCSchemeEnvrionmentVariables+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        .init(
+            variable: "BUILD_WORKSPACE_DIRECTORY",
+            value: "$(SRCROOT)",
+            enabled: true
+        ),
+        .init(
+            variable: "BUILD_WORKING_DIRECTORY",
+            // This is a poor substitute for the working directory.
+            // Preferably, it would be $(PWD) or something similar.
+            // Unfortunately, none of the following worked:
```
Comment (brentleyjones) [inline tools/generator/src/XCSchemeEnvrionmentVariables+Extensions.swift:19]: We can remove this last line. Xcode doesn't have a concept of a working directory. Schemes can have a working directory though, and they default to something inside of derives data. Maybe we should set this value to whatever that defaults to, or change the working directory to be SRCROOT?

Comment (cgrindel) [reply tools/generator/src/XCSchemeEnvrionmentVariables+Extensions.swift:19]: I see [a way to specify that you want to set a custom working directory](https://github.com/tuist/XcodeProj/blob/main/Sources/XcodeProj/Scheme/XCScheme%2BLaunchAction.swift#L45). Unfortunately, I don't see a way to set the actual custom working directory. 

When I set it in the Xcode UI, I see the following in the scheme file:

```xml
<LaunchAction
      useCustomWorkingDirectory = "YES"
      customWorkingDirectory = "/Users/chuckgrindel"
```

Do you know if there is a macro for the derives data location? I will poke around to try and find one.

Comment (cgrindel) [reply tools/generator/src/XCSchemeEnvrionmentVariables+Extensions.swift]: It looks like `BUILD_PRODUCTS_DIR` evaluates to the directory that Xcode uses as the current working directory.

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**